File: chapter_08/subset.go
package main

import "fmt"

func isSubset[T comparable](arr1, arr2 []T) bool {
	var largeArray, smallArray []T
	if len(arr1) > len(arr2) {
		largeArray = arr1
		smallArray = arr2
	} else {
		largeArray = arr2
		smallArray = arr1
	}

	for i := 0; i < len(smallArray); i++ {
		foundMatch := false

		for j := 0; j < len(largeArray); j++ {
			if smallArray[i] == largeArray[j] {
				foundMatch = true
				break
			}
		}
		if !foundMatch {
			return false
		}
	}
	return true
}

func main() {
	arrLarge := []int{1, 3, 4, 5, 6}
	arrSmall := []int{3, 4}

	res := isSubset(arrLarge, arrSmall)
	fmt.Println(res)
}

File: chapter_08/subset.js
function isSubset(array1, array2) {
  let largerArray;
  let smallerArray;

  if (array1.length > array2.length) {
    largerArray = array1;
    smallerArray = array2;
  } else {
    largerArray = array2;
    smallerArray = array1;
  }

  // Iterate through smaller array
  for (let i = 0; i < smallerArray.length; i++) {
    let foundMatch = false;

    for (let j = 0; j < largerArray.length; j++) {
      if (smallerArray[i] === largerArray[j]) {
        foundMatch = true;
        break;
      }
    }
    if (foundMatch === false) {
      return false;
    }
  }
  return true;
}

let arrLarge = [1, 3, 4, 5, 6];
let arrSmall = [3, 4];

console.log(isSubset(arrSmall, arrLarge));
